src/lyapunovspectrum.jl

```diff
@@ -86,16 +96,54 @@ function modifiedGS!(A, Q, R, aⱼ, qᵢ, vⱼ)
     return nothing
 end
 
-function variational_eqs!(t::Taylor1{T}, x::Vector{Taylor1{U}},
-        dx::Vector{Taylor1{U}}, jac::Array{Taylor1{U},2}, jacobianfunc! =Nothing) where {T<:Real, U<:Number}
-    # Dimensions of phase-space: dof
-    nx = length(x)
-    dof = size(jac, 1)
-    if jacobianfunc! != Nothing
-        # Stability matrix
-        jacobianfunc!(jac, t, x)
+"""
+    lyap_jetcoeffs!(t, x, dx, jac)
+
+Similar to [`jetcoeffs!`](@ref) for the calculation of the Lyapunov spectrum.
+Updates *only* the elements of `x` which correspond to the solution of the 1st-order
+variational equations \$\\dot{\\xi}=J \\cdot \\xi\$, where \$J\$ is the Jacobian
+matrix, i.e., the linearization of the equations of motion. `jac` is the Taylor
+expansion of \$J\$ wrt the independent variable, around the current initial
+condition. Calling this method assumes that `jac` has been computed previously
+using [`stabilitymatrix!`](@ref).
+
+"""
+function lyap_jetcoeffs!(t::Taylor1{T}, x::AbstractVector{Taylor1{S}},
+        dx::AbstractVector{Taylor1{S}}, jac::Matrix{Taylor1{S}}) where {T <: Real, S <: Number}
+    order = t.order
+    nlyaps = size(jac, 1) # number of Lyapunov exponents
+    temp_002 = Array{eltype(x)}(undef, nlyaps, nlyaps, nlyaps)
+    temp_001 = Array{Taylor1{S}}(undef, size(temp_002))
+    @__dot__ temp_001 = Taylor1(zero(S), order)
+    for j = 1:nlyaps
```

Comment:
If I understand correctly the algorithm, it actually performs explicitly the matrix multiplication to integrate the variationals, with a lot of care in the indices.  What about assuming that the indices of `x` and `dx` begin at 1? In that case, `lyap_jetcoeffs!` would be called via a view or something similar? My point is that such a change would make simpler the indexing, actually saving some operations.

Reply:
Yes, exactly, the matrix multiplication was implemented explicitly in order to avoid the use of `reshape` and also to perform operations in-place. About the indexing, `x` and `dx` are already assumed to begin at 1 and called with views over `x` and `dx` (see, for example, `lyap_jetcoeffs!` call in `lyap_taylorstep!`, line 174 of `src/lyapunovspectrum.jl`). The ugly indexing (with `1:nlyaps`, etc.) comes from trying to implement the matrix multiplication of, say, `dx = jac*x` where `jac` is a square matrix with size `(nlyaps, nlyaps)` but `x` and `dx` are actually 1-dim arrays (i.e, vectors) with size `nlyaps*nlyaps`

Reply:
Thanks for clarifying this; perhaps you should add some comments in the program to remind us.

I would then suggest to define the abbreviations (change the names at your will):
```jl
indx1 = nlyaps * (j - 1) + i  # after line 119, and also after 134 
indx2 = nlyaps * (j - 1) + k  # after line 121, and also after 136
```
and use them in the lines following those definitions. This may lead to some insignificant performance improvement, and clarity.